src/core/server/logging/appenders/rolling_file/strategies/numeric/pattern_matcher.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const escape = (string: string) => string.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&');
```

Comment:
**optional nit** I know that lodash is sometimes overused, but they offer [`_.escapeRegExp()`](https://lodash.com/docs/4.17.15#escapeRegExp) so that you don't have to re-invent or re-test this.